Startup issue

I got such an ERROR, and the application does not make progress anymore.

<details>
<summary>Click to see the stack trace!</summary>

```
2022-05-24 12:06:45,008 [main]          INFO  o.a.explorer.Main$                     - Build info: name: explorer-backend-app, scalaVersion: 2.13.3, sbtVersion: 1.6.2, commitId: 76d0974ca8d0a1f6a0a61d22ce7ee281711a4def, branch: leman, releaseVersion: 1.7.0-leman0

2022-05-24 12:06:45,351 [-dispatcher-6] INFO  a.e.s.Slf4jLogger                      - Slf4jLogger started

2022-05-24 12:06:45,916 [main]          INFO  c.z.h.HikariDataSource                 - db.db - Starting...

2022-05-24 12:06:45,933 [main]          INFO  c.z.h.HikariDataSource                 - db.db - Start completed.

2022-05-24 12:06:46,076 [main]          DEBUG i.m.c.u.i.l.InternalLoggerFactory      - Using SLF4J as the default logging framework

2022-05-24 12:06:46,184 [-dispatcher-6] ERROR o.a.e.c.AsyncReloadingCache            - Failed to reload cache

java.lang.RuntimeException: org.postgresql.util.PSQLException: ERROR: relation "transactions" does not exist

  Position: 22

at org.alephium.explorer.persistence.DBRunner$$anonfun$run$3.applyOrElse(DBRunner.scala:55)

at org.alephium.explorer.persistence.DBRunner$$anonfun$run$3.applyOrElse(DBRunner.scala:54)

at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:453)

at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:63)

at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:100)

at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)

at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:94)

at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:100)

at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:49)

at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:48)

at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)

at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)

at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)

at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)

at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)

Caused by: org.postgresql.util.PSQLException: ERROR: relation "transactions" does not exist

  Position: 22

at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2533)

at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2268)

at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:313)

at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:448)

at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:369)

at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:159)

at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:148)

at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)

at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)

at slick.jdbc.StatementInvoker.results(StatementInvoker.scala:39)

at slick.jdbc.StatementInvoker.iteratorTo(StatementInvoker.scala:22)

at slick.jdbc.Invoker.first(Invoker.scala:30)

at slick.jdbc.Invoker.first$(Invoker.scala:29)

at slick.jdbc.StatementInvoker.first(StatementInvoker.scala:16)

at slick.jdbc.JdbcActionComponent$QueryActionExtensionMethodsImpl$$anon$3.run(JdbcActionComponent.scala:224)

at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:28)

at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:25)

at slick.basic.BasicBackend$DatabaseDef$$anon$3.liftedTree1$1(BasicBackend.scala:276)

at slick.basic.BasicBackend$DatabaseDef$$anon$3.run(BasicBackend.scala:276)

at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)

at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)

at java.base/java.lang.Thread.run(Unknown Source)

Exception in thread "main" java.lang.RuntimeException: org.postgresql.util.PSQLException: ERROR: relation "transactions" does not exist

  Position: 22

at org.alephium.explorer.persistence.DBRunner$$anonfun$run$3.applyOrElse(DBRunner.scala:55)

at org.alephium.explorer.persistence.DBRunner$$anonfun$run$3.applyOrElse(DBRunner.scala:54)

at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:453)

at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:63)

at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:100)

at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)

at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:94)

at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:100)

at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:49)

at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:48)

at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)

at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)

at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)

at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)

at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)

Caused by: org.postgresql.util.PSQLException: ERROR: relation "transactions" does not exist

  Position: 22

at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2533)

at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2268)

at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:313)

at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:448)

at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:369)

at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:159)

at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:148)

at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)

at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)

at slick.jdbc.StatementInvoker.results(StatementInvoker.scala:39)

at slick.jdbc.StatementInvoker.iteratorTo(StatementInvoker.scala:22)

at slick.jdbc.Invoker.first(Invoker.scala:30)

at slick.jdbc.Invoker.first$(Invoker.scala:29)

at slick.jdbc.StatementInvoker.first(StatementInvoker.scala:16)

at slick.jdbc.JdbcActionComponent$QueryActionExtensionMethodsImpl$$anon$3.run(JdbcActionComponent.scala:224)

at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:28)

at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:25)

at slick.basic.BasicBackend$DatabaseDef$$anon$3.liftedTree1$1(BasicBackend.scala:276)

at slick.basic.BasicBackend$DatabaseDef$$anon$3.run(BasicBackend.scala:276)

at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)

at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)

at java.base/java.lang.Thread.run(Unknown Source)
```
</details>

--- a/app/src/main/scala/org/alephium/explorer/Application.scala
+++ b/app/src/main/scala/org/alephium/explorer/Application.scala
@@ -16,7 +16,7 @@
 
 package org.alephium.explorer
 
-import scala.concurrent.{Await, ExecutionContext, Future, Promise}
+import scala.concurrent.{ExecutionContext, Future, Promise}
 import scala.concurrent.duration._
 
 import akka.actor.ActorSystem
@@ -31,6 +31,7 @@ import org.alephium.explorer.cache.{BlockCache, TransactionCache}
 import org.alephium.explorer.persistence.DBInitializer
 import org.alephium.explorer.persistence.dao._
 import org.alephium.explorer.service._
+import org.alephium.explorer.util.FutureUtil._
 import org.alephium.explorer.util.Scheduler
 import org.alephium.protocol.model.NetworkId
 import org.alephium.util.Duration
@@ -49,11 +50,9 @@ class Application(host: String,
                                                databaseConfig: DatabaseConfig[PostgresProfile])
     extends StrictLogging {
 
-  //TOTO: Temporary placeholder for executing services returning a Future.
-  //      This will eventually be removed and the all services will get started
-  //      asynchronously via the start function.
-  def awaitService[T](f: => Future[T]): T =
-    Await.result(f, 5.seconds)
+  if (!readOnly) {
+    DBInitializer.initialize().await(10.seconds)
+  }
 
   implicit val scheduler: Scheduler =
     Scheduler(s"${classOf[Application].getSimpleName} scheduler")
@@ -65,7 +64,7 @@ class Application(host: String,
     BlockCache()
 
   implicit val transactionCache: TransactionCache =
-    awaitService(TransactionCache())
+    TransactionCache().await(5.seconds)
 
   //Services
   implicit val blockFlowClient: BlockFlowClient =
@@ -121,11 +120,7 @@ class Application(host: String,
 
   private def startTasksForReadWriteApp(): Future[Unit] = {
     if (!readOnly) {
-      for {
-        _ <- DBInitializer.initialize()
-        _ <- startSyncService()
-      } yield ()
-
+      startSyncService()
     } else {
       Future.successful(())
     }

--- a/app/src/main/scala/org/alephium/explorer/util/FutureUtil.scala
+++ b/app/src/main/scala/org/alephium/explorer/util/FutureUtil.scala
@@ -14,18 +14,18 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the library. If not, see <http://www.gnu.org/licenses/>.
 
-package org.alephium.explorer.benchmark
+package org.alephium.explorer.util
 
 import scala.concurrent.{Await, Future}
 import scala.concurrent.duration._
 
-object BenchmarkUtils {
+object FutureUtil {
 
   implicit class AwaitUtils[T](f: => Future[T]) {
 
     //Convenience function to wait on a future
     @SuppressWarnings(Array("org.wartremover.warts.DefaultArguments"))
-    def await(seconds: FiniteDuration = 5.seconds): T =
+    @inline def await(seconds: FiniteDuration = 5.seconds): T =
       Await.result(f, seconds)
 
   }

